test/shared/src/test/scala/zio/test/TestAspectSpec.scala

```diff
@@ -132,6 +143,32 @@ object TestAspectSpec extends DefaultRuntime {
       failed(spec)
     }
 
+  def timeoutMakesTestsFailAfterGivenDuration: Future[Boolean] =
+    unsafeRunToFuture {
+      val spec = (testM("timeoutMakesTestsFailAfterGivenDuration") {
+        assertM(ZIO.never *> ZIO.unit, equalTo(()))
+      }: ZSpec[Live[Clock], Any, String, Any]) @@ timeout(100.millis)
+
+      testExecutionFailedWith(
+        spec,
+        cause => cause.isInstanceOf[TestTimeoutException] && cause.getMessage() == "Timeout of 100 ms exceeded."
```

Comment:
You could use a pattern match here if you want to avoid the `isInstanceOf` check.